src/db/options.rs

```diff
@@ -136,3 +136,11 @@ pub struct ListCollectionsOptions {
     #[serde(serialize_with = "bson_util::serialize_batch_size", rename = "cursor")]
     pub batch_size: Option<u32>,
 }
+
+#[derive(Debug, Default, TypedBuilder, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ListDatabasesOptions {
+    // Determines which databases to return based on the user's access privileges.
```

Comment:
`//` (with two slashes) denotes a comment, whereas `///` (with three) denotes a docstring that will be rendered in the documentation. So let's add a slash here to make sure the latter happens.

Also, we should probably mention that this option is only supported in MongoDB 4.0.5+.

Lastly, let's add a docstring to the options type itself that links to the methods that use it. Check out other options structs for the format (e.g. [`CreateCollectionOptions`](https://github.com/mongodb/mongo-rust-driver/blob/master/src/db/options.rs#L31)).